ecosystem/gstreamer_plugin/gst_mtl_st40_rx.c

```diff
@@ -372,67 +385,144 @@ static void* gst_mtl_st40_rx_get_mbuf_with_timeout(Gst_Mtl_St40_Rx* src,
   return mbuf;
 }
 
+static struct st40_rfc8331_payload_hdr* gst_mtl_st40_rx_shift_payload_hdr(
+    struct st40_rfc8331_payload_hdr* payload_hdr, int udw_size) {
+  gint package_size;
+  gint payload_len;
+
+  package_size = ((WORD_10_BIT_ALIGN + udw_size) * USER_DATA_WORD_BIT_SIZE) / BYTE_SIZE;
+  payload_len = sizeof(struct st40_rfc8331_payload_hdr) -
+                (package_size % WORD_10_BIT_ALIGN) + package_size;
+
+  return (struct st40_rfc8331_payload_hdr*)((uint8_t*)payload_hdr + payload_len);
+}
+
+static GstFlowReturn gst_mtl_st40_rx_check_parity(
+    const struct st40_rfc8331_payload_hdr* payload_hdr) {
+  if (!st40_check_parity_bits(payload_hdr->second_hdr_chunk.did)) {
+    GST_ERROR("Parity check failed for DID");
+    return GST_FLOW_ERROR;
+  }
+  if (!st40_check_parity_bits(payload_hdr->second_hdr_chunk.sdid)) {
+    GST_ERROR("Parity check failed for SDID");
+    return GST_FLOW_ERROR;
+  }
+  if (!st40_check_parity_bits(payload_hdr->second_hdr_chunk.data_count)) {
+    GST_ERROR("Parity check failed for Data Count");
+    return GST_FLOW_ERROR;
+  }
+  return GST_FLOW_OK;
+}
+
 static GstFlowReturn gst_mtl_st40_rx_fill_buffer(Gst_Mtl_St40_Rx* src, GstBuffer** buffer,
                                                  void* usrptr) {
   struct st40_rfc8331_rtp_hdr* hdr;
-  struct st40_rfc8331_payload_hdr *payload_hdr, payload_hdr_swapped;
+  struct st40_rfc8331_payload_hdr* payload_hdr;
   GstMapInfo dest_info;
-  guint16 data, fill_size;
-  gint udw_size;
+  guint16 data;
+  gint udw_size, anc_count;
+  guint buffer_size = 0, meta_offset;
 
   hdr = (struct st40_rfc8331_rtp_hdr*)usrptr;
+  anc_count = hdr->first_hdr_chunk.anc_count;
```

Comment:
anc_count is saved as 8 bits of uint32_t. I think it is pointless to save it in signed int and check if it is <0. save it as uint and assume it is always >=0